contracts/impl/TaskImpl.sol

```diff
@@ -146,22 +126,47 @@ library TaskImpl {
         message = EnigmaCommon.appendMessage(message, hex"01");
         bytes32 msgHash = keccak256(message);
         require(msgHash.recover(_sig) == state.workers[msg.sender].signer, "Invalid signature");
+    }
 
-        // Set the secret contract's attributes in registry
+    function deploySecretContractImpl(EnigmaState.State storage state, bytes32 _taskId, bytes32 _preCodeHash,
+        bytes32 _codeHash, bytes32 _initStateDeltaHash, bytes memory _optionalEthereumData,
+        address _optionalEthereumContractAddress, uint64 _gasUsed, bytes memory _sig)
+    public
+    {
         EnigmaCommon.SecretContract storage secretContract = state.contracts[_taskId];
-        secretContract.owner = task.sender;
-        secretContract.preCodeHash = _preCodeHash;
-        secretContract.codeHash = _codeHash;
-        secretContract.status = EnigmaCommon.SecretContractStatus.Deployed;
-        secretContract.stateDeltaHashes.push(_initStateDeltaHash);
-        state.scAddresses.push(_taskId);
+        EnigmaCommon.TaskRecord storage task = state.tasks[_taskId];
+        verifyDeployReceipt(state, _taskId, _codeHash, _initStateDeltaHash, _optionalEthereumData,
+            _optionalEthereumContractAddress, _gasUsed, msg.sender, _sig);
+        task.proof = _sig;
 
         if (_optionalEthereumContractAddress != address(0)) {
-            (bool success,) = _optionalEthereumContractAddress.call(_optionalEthereumData);
-            require(success, "Ethereum call failed");
+            require(gasleft() > 150000, "Not enough gas from worker for eth callback");
```

Comment:
where do these hardcoded values for `150000` and `100000` come from?

Reply:
The next anticipated iteration of this will likely use a combination of the user specifying eth callback gas limit upon task creation and ENG/ETH decentralized price feed conversion rates to translate unused ENG gas to ETH gas limit for the callback and used ETH gas back to ENG to adjust balances. 

For the time being until that more robust solution is implemented (which will require some function signature reordering and restructuring to avoid stack issues in the EVM and related updates to other components of our network), I determined that the remainder of the `commitReceipt` function after the callback requires no more than 100000 gas. There is a corner case here however where the worker can submit just barely above 100000 gas and it fails in the callback, so wanted to make sure they pass at least 150000, which leaves ~50000 for the callback. 

Overall, this is I believe a temporary solution that works, but will implement the more robust solution asap. @fredfortier @elichai

Reply:
Thanks